### PR TITLE
Update kzg interface

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Resolves https://github.com/sigp/lighthouse/issues/4076

## Proposed Changes

Updates the kzg interface for freeing the blobs.

### CODE REVIEW

**Comment by pawanjay176** [comment]
@jimmygchen Not able to request you as reviewer for some reason.

File: crypto/kzg/Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ serde_derive = "1.0.116"
 eth2_serde_utils = "0.1.1"
 hex = "0.4.2"
 eth2_hashing = "0.3.0"
-c-kzg = {git = "https://github.com/ethereum/c-kzg-4844", rev = "69f6155d7524247be9d3f54ab3bfbe33a0345622" }
+c-kzg = {git = "https://github.com/ethereum/c-kzg-4844", branch = "main" }
```
Comment (realbigsean) [inline crypto/kzg/Cargo.toml]: we should still probably pin this to a commit right?

**Review by realbigsean — APPROVED** [review]

File: beacon_node/beacon_chain/src/kzg_utils.rs [hunk 2]
```diff
@@ -1,54 +1,57 @@
...
+    kzg.verify_blob_kzg_proof(
+        ssz_blob_to_crypto_blob::<T>(blob),
+        kzg_commitment,
+        kzg_proof,
     )
 }
 
-pub fn compute_aggregate_kzg_proof<T: EthSpec>(
+/// Validate a batch of blob-commitment-proof triplets from multiple `BlobSidecars`.
+pub fn validate_blobs_sidecar<T: EthSpec>(
```
Comment (jimmygchen) [inline beacon_node/beacon_chain/src/kzg_utils.rs]: nit: 

```suggestion
pub fn validate_blob_sidecars<T: EthSpec>(
```

Comment (pawanjay176) [reply beacon_node/beacon_chain/src/kzg_utils.rs]: Changed it to `validate_blobs` to match the fork choice spec.

File: crypto/kzg/Cargo.toml (hunk 1 above)
Comment (pawanjay176) [reply crypto/kzg/Cargo.toml]: Done!

**Comment by divagant-martian** [comment]
after this Pr the main branch no longer checks.
Doing `cargo clean` and then `cargo check` produces
```bash
beacon_node/beacon_chain/src/beacon_chain.rs:4763:28: error[E0425]: cannot find function `compute_aggregate_kzg_proof` in module `kzg_utils`
beacon_node/beacon_chain/src/beacon_chain.rs:4777:24: error[E0425]: cannot find function `validate_blobs_sidecar` in module `kzg_utils`
beacon_node/beacon_chain/src/blob_verification.rs:138:20: error[E0425]: cannot find function `validate_blobs_sidecar` in module `kzg_utils`
```

**Comment by jimmygchen** [comment]
Thanks @pawanjay176 and @divagant-martian 
I'll update my branch with @pawanjay176 's changes and this should fix the compilation error.

**Comment by pawanjay176** [comment]
Sorry for borking the base branch :sweat_smile:  Didn't realise we still had all the kzg computation from coupling days.

**Pull request merged**